doc/requirements/srsd.rst

```diff
@@ -2,10 +2,126 @@
+#. The ``nuclearDataIO`` package is used to read and write nuclear data files which are used as input to global flux solvers, and eventually input into safety calculations via reactivity coefficients. Therefore, the ``nuclearDataIO`` package is considered high risk.
+
+
+--------------------
+Applicable Documents
+--------------------
+
+..
+   TODO: Do this by topic
+
+
 -----------------------
 Functional Requirements
 -----------------------
 
+.. req:: The database shall maintain fidelity of data.
+   :id: REQ_DB_FIDELITY
+   :status: needs implementation, needs test
+
+The database shall faithfully represent the possessed information and not alter its contents, retrieving the data exactly as input.
+
+.. req:: The database shall allow case restarts.
+   :id: REQ_DB_RESTARTS
+   :status: needs implementation, needs test
+
+The state information representing as near as possible the entirety of the run when the database write was executed, shall be retrievable to restore the previous case to a particular point in time for further use by analysts.
+
+.. req:: The database shall accept all pythonic primitive data types.
+   :id: REEQ_DB_PRIMITIVES
+   :status: needs implementation, needs test
+
+Given the ubiquity of Python's ``None`` the database shall support its inclusion as a valid entry for data. There will be no support for any abstract data type beyond None.
+
+.. req:: The report package shall maintain data fidelity.
+   :id: REQ_REPORT_FIDELITY
+   :status: implemented, needs more tests
+
+The report package shall not modify or subvert data integrity as it reports the information out to the user.
+
+
+..
+   TODO: blueprints need some interface and I/O reqs
+
+.. req:: The settings package shall not accept ambiguous setting definitions.
+   :id: REQ_SETTINGS_UNAMBIGUOUS
+   :status: implemented, needs more tests
+
+Settings defined in the system must have both the intended data type and default value defined, or it is considered incomplete and therefore invalid. Additionally the system shall not accept multiple definitions of the same name.
+
+TODO: This may be tested by a unit test loading in duplicate setting definitions or cases where a definition does not provide adequate details.
+
+   .. req:: Settings shall have unique, case-insensitive names.
+      :id: REQ_SETTINGS_UNAMBIGUOUS_NAME
+      :status: implemented, needs more tests
+
+   No two settings may share names.
+
+   TODO: This may be tested by a unit test loading two similar names
+
+   .. req:: Settings shall not allow dynamic typing.
+      :id: REQ_SETTINGS_UNAMBIGUOUS_TYPE
+      :status: implemented, needs more tests
+
+   Settings shall exist exclusively as a well-defined datatype, as chosen by the setting definition.
```

Comment:
I don't know if "data type" or "datatype" is preferred, but I just noticed in this setting it is used inconsistently. Maybe checking for this and other common inconsistent wording (filepath, timestep, etc) is important? Or not...up to you.

Reply:
Okay, so Wikipedia says it's two words: https://en.wikipedia.org/wiki/Data_type

Thanks!

Reply:
One thing is always true: It's always the opposite of that I think it is. 😆